### PR TITLE
Added a new parameter to FileCopier to allow copying of emtpy dirs

### PR DESCRIPTION
**Pull request by claasd**
I need to copy over empty directories.
I added a new parameter that allows this, please review.

### CODE REVIEW

File: conans/test/functional/file_copier_test.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import unittest
 import os
+from turtledemo.chaos import f
```
Comment (memsharded) [inline conans/test/functional/file_copier_test.py]: Unrelated import?

**Comment by claasd** [comment]
I have no idea how that import got there ...

File: conans/test/functional/file_copier_test.py (hunk 1 above)
Comment (claasd) [inline conans/test/functional/file_copier_test.py]: I have no idea how that import got there ...
probably did some fancy key shortcut without noticing

**Comment by memsharded** [comment]
Just curiosity, what is your use case for needing to copy empty dirs?

**Comment by claasd** [comment]
Our previously used package manager did pack empty dirs by when using `*`.
We use conan to package tests, to be able to execute the exact same tests on different machines. Some of these tests need a predefined file structure that contains empty folders (kind of legacy behavior, but currently not changeable).

**Comment by memsharded** [comment]
Ok, thanks very much, understood. In my opinion sounds too much like an extremely specific use-case for some legacy thing, not very sure about the addition and value for other users. As a quick idea, I guess that if you need such empty folders, why not just create them in the ``package_info()`` method? Something like:

```python
from conans.util.files import mkdir
def package_info(self):
    mkdir(os.path.join(self.package_folder, "folder1"))
    mkdir(os.path.join(self.package_folder, "folder2"))
```

The conan ``mkdir`` will not fail if the folder already exists. Please let us know if this could be a solution too.

Just another side issue. I think this feature (if approved) would deserve an integration test with upload/download to remote (testing remote), because I think there are chances that such folders will be lost in the package zip & unzip while transferring to remotes. Lets wait for @lasote feedback.

**Comment by claasd** [comment]
Ok, I need to remind myself to first ask how it could be achieved before implementing it and starting a pull request :-) 
I did not know that I could create folders in the package_info method.
Let my try it out and get back to you, probably you can discard this pull request.

**Comment by claasd** [comment]
Your solution did solve same issue I addressed.
For now, both do not solve my use case, as I need to import the file structure using a conanfile.txt (and thus, the created folder was not copied over).

Probably I need to address this differently.

**Comment by memsharded** [comment]
It is absolutely compulsory to use the ``conanfile.txt``, or could you use a ``conanfile.py`` for consuming? Because maybe in that case you could just create the folders locally in the ``imports()`` method. Just let us know how it goes, and how can we help in finding a solution to your problem.

**Pull request closed without merging**